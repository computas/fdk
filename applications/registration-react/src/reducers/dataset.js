import {
  DATASET_REQUEST,
  DATASET_SUCCESS,
  DATASET_FAILURE
} from '../constants/ActionTypes';

export default function dataset(
  state = { isFetching: false, result: null },
  action
) {
  switch (action.type) {
    case DATASET_REQUEST: {
      return {
        ...state,
        isFetching: true
      };
    }
    case DATASET_SUCCESS: {
      return {
        ...state,
        isFetching: false,
        result: action.response.data
      };
    }
    case DATASET_FAILURE: {
      return {
        ...state,
        isFetching: false,
        result: null
      };
    }
    default:
      return state;
  }
}
