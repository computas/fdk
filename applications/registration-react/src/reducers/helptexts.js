import {
  HELPTEXTS_REQUEST,
  HELPTEXTS_SUCCESS,
  HELPTEXTS_FAILURE
} from '../constants/ActionTypes';

export default function dataset(
  state = { isFetchingHelptext: false, helptextItems: null },
  action
) {
  switch (action.type) {
    case HELPTEXTS_REQUEST: {
      return {
        ...state,
        isFetchingHelptext: true
      };
    }
    case HELPTEXTS_SUCCESS: {
      const objFromArray = action.response.data.reduce(
        (accumulator, current) => {
          accumulator[current.id] = current; // eslint-disable-line no-param-reassign
          return accumulator;
        },
        {}
      );
      return {
        ...state,
        isFetchingHelptext: false,
        helptextItems: objFromArray
      };
    }
    case HELPTEXTS_FAILURE: {
      return {
        ...state,
        isFetchingHelptext: false,
        helptextItems: null
      };
    }
    default:
      return state;
  }
}
