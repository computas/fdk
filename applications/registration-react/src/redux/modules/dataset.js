import {
  DATASET_REQUEST,
  DATASET_SUCCESS,
  DATASET_FAILURE
} from '../../constants/ActionTypes';

export default function dataset(
  state = { isFetching: false, result: null },
  action
) {
  switch (action.type) {
    case DATASET_REQUEST: {
      return {
        ...state,
        isFetching: true
      };
    }
    case DATASET_SUCCESS: {
      return {
        ...state,
        isFetching: false,
        result: action.payload
      };
    }
    case DATASET_FAILURE: {
      return {
        ...state,
        isFetching: false,
        result: null
      };
    }
    default:
      return state;
  }
}
