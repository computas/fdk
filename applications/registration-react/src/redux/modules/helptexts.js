import _ from 'lodash';
import { fetchActions } from '../fetchActions';

const HELPTEXTS_REQUEST = 'HELPTEXTS_REQUEST';
const HELPTEXTS_SUCCESS = 'HELPTEXTS_SUCCESS';
const HELPTEXTS_FAILURE = 'HELPTEXTS_FAILURE';

function shouldFetch(metaState) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      (metaState.lastFetch || 0) < Date.now() - threshold)
  );
}

export function fetchHelptextsIfNeeded() {
  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['helptexts', 'meta'])) &&
    dispatch(
      fetchActions('/reference-data/helptexts', [
        HELPTEXTS_REQUEST,
        HELPTEXTS_SUCCESS,
        HELPTEXTS_FAILURE
      ])
    );
}

const initialState = {};

export default function helptexts(state = initialState, action) {
  switch (action.type) {
    case HELPTEXTS_REQUEST: {
      return {
        ...state,
        meta: {
          isFetching: true,
          lastFetch: null
        }
      };
    }
    case HELPTEXTS_SUCCESS: {
      const objFromArray = action.payload.reduce((accumulator, current) => {
        accumulator[current.id] = current; // eslint-disable-line no-param-reassign
        return accumulator;
      }, {});
      return {
        ...state,
        helptextItems: objFromArray,
        meta: {
          isFetching: false,
          lastFetch: Date.now()
        }
      };
    }
    case HELPTEXTS_FAILURE: {
      return {
        ...state,
        helptextItems: null,
        meta: {
          isFetching: false,
          lastFetch: null
        }
      };
    }
    default:
      return state;
  }
}
