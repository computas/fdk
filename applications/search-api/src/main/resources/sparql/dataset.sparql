PREFIX dcat: <http://www.w3.org/ns/dcat#>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX owl: <http://www.w3.org/TR/owl-time/>
PREFIX adms: <http://www.w3.org/ns/adms#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>
PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
PREFIX dcatno: <http://difi.no/dcatno#>
prefix vcard: <http://www.w3.org/2006/vcard/ns#>
PREFIX er: <http://data.brreg.no/meta/>
prefix xsd: <http://www.w3.org/2001/XMLSchema#>
prefix schema: <http://schema.org/>

DESCRIBE ?dataset ?publisher ?contact ?distribution ?temporal ?spatial ?nace ?fadr ?padr ?sektor
WHERE {
    ?dataset a dcat:Dataset;
    dct:publisher ?publisher.

    OPTIONAL {?dataset dcat:contactPoint ?contact}
    OPTIONAL {?dataset dcat:distribution ?distribution }
    OPTIONAL {?dataset dcat:contactPoint ?contact}
    OPTIONAL {?dataset dct:temporal ?temporal}
    OPTIONAL {?dataset dct:spatial ?spatial}

    OPTIONAL {?publisher er:naeringskode1 ?nace}
    OPTIONAL {?publisher er:forretningsadresse ?fadr}
    OPTIONAL {?publisher er:postadresse ?padr}
    OPTIONAL {?publisher er:institusjonellSektorkode ?sektor}

    FILTER (?dataset = <%s> )
}