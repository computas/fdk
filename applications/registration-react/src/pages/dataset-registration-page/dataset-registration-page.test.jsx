import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import shallowWithStore from '../../../test/shallowWithStore';
import RegDatasetComponent, { RegDataset } from './dataset-registration-page';
import dataset from '../../../test/fixtures/datasets';
import helptexts from '../../../test/fixtures/helptext';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  dispatch = jest.fn();
  defaultProps = {
    dispatch,
    helptextItems: helptexts.helptextItems,
    isFetching: false
  };
  wrapper = shallow(<RegDataset {...defaultProps} />);
});

test('should render RegDataset correctly', () => {
  expect(wrapper).toHaveLength(1);
  expect(wrapper).toMatchSnapshot();
});

test('should render FormAccessRightsSchema correctly', () => {
  const testState = {
    dataset: dataset.datasetItems._embedded.datasets[0],
    helptexts,
    form: null
  };
  const store = createMockStore(testState);
  wrapper = shallowWithStore(<RegDatasetComponent />, store);
  expect(wrapper).toHaveLength(1);
});
