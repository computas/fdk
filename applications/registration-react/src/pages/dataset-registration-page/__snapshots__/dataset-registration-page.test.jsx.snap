// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RegDataset correctly 1`] = `
<div
  className="container"
>
  <div
    className="row mb-2 mb-md-5"
  >
    <div
      className="col-md-2"
    >
      <i
        className="fa fa-arrow-left mr-2"
      />
      <Link
        className="fdk-text-size-small fdk-color1 font-weight-light"
        replace={false}
        to=""
      >
        Tilbake til katalogen
      </Link>
    </div>
    <div
      className="col-md-2"
    />
  </div>
</div>
`;
